src/components/SettingsModal/SettingsStorage.js

```diff
@@ -31,9 +33,34 @@ export function getDefaultSettings() {
  */
 export async function getSettingsValue(key) {
   const value = await investSettingsStore.getItem(key);
+  if (!value) {
+    return getDefaultSettings()[key];
+  }
   return value;
 }
 
+/** Getter function for entire contents of store.
+ *
+ * @returns {Object} - key: value pairs of settings
+ */
+export async function getAllSettings() {
+  try {
+    const promises = [];
+    const keys = Object.keys(getDefaultSettings());
+    keys.forEach((key) => {
+      promises.push(getSettingsValue(key));
+    });
+    const values = await Promise.all(promises);
```

Comment:
Nice, I like workflow of requesting all the values and then awaiting responses for all the promises.

Reply:
I recall being a little surprised there was no localForage API for retrieving the whole db, so this seems to be the conventional way to do it, one key at a time.